src/boost-api/endpoint_handlers.cpp

```diff
@@ -1,20 +1,152 @@
+        const std::string& requestBody = req.body();
+
+        boost::json::value jsonData;
+        try {
+            boost::json::error_code ec;
+            jsonData = boost::json::parse(requestBody, ec);
+            if (ec)
+                throw std::runtime_error("Failed to parse JSON: " + ec.message());
+        } catch (const std::exception& e) {
+            res.result(status::bad_request);
+            res.set(field::content_type, "text/plain");
+            res.body() = "Failed to parse JSON: " + std::string(e.what());
+            return {};
+        }
+
+        return jsonData;
+    }
+
+    void handle_insert_trajectories_into_trajectory_table(const request<string_body> &req, response<string_body> &res) {
+        boost::json::value jsonData = get_json_from_request_body(req, res);
+        if (jsonData.is_null())
+            return;
+
+        std::vector<data_structures::Trajectory> all_trajectories{};
+        db_table db_table{};
+
+        try {
+            auto jsonObject = jsonData.as_object();
+            for (auto & it : jsonObject) {
+                const auto &table_key = it.key();
+                const auto &table_value = it.value().as_object();
+                db_table = table_key == "original" ? db_table::original_trajectories
+                                                   : db_table::simplified_trajectories;
+
+                for (const auto & jt : table_value) {
+                    const auto &trajectory_id = jt.key();
+                    const auto &trajectoryData = jt.value().as_object();
+
+                    data_structures::Trajectory trajectory;
+                    trajectory.id = std::stoi(trajectory_id);
+
+                    for (const auto & kt : trajectoryData) {
+                        const auto &location_order = kt.key();
+                        const auto &location_data = kt.value().as_object();
+
+                        data_structures::Location location{};
+                        location.order = std::stoi(location_order);
+                        location.timestamp = location_data.at("timestamp").as_double();
+                        location.longitude = location_data.at("longitude").as_double();
+                        location.latitude = location_data.at("latitude").as_double();
+
+                        trajectory.locations.emplace_back(location);
+                    }
+                    all_trajectories.emplace_back(std::move(trajectory));
+                }
+            }
+            trajectory_manager::insert_trajectories_into_trajectory_table(all_trajectories, db_table);
+        } catch (const std::exception &e) {
+            res.result(status::bad_request);
+            res.set(field::content_type, "text/plain");
+            res.body() = "Error processing JSON data: " + std::string(e.what());
+        }
+    }
+
+    void handle_load_trajectories_into_rtree(const request<string_body> &req, response<string_body> &res) {
```

Comment:
Vi får desværre ikke brug for denne funktion længere, da vi ikke har et r-tree, men bare et index, som opfører sig som et r-tree i den nye db.

Reply:
Removed